### PR TITLE
Sign deb packages

### PR DESCRIPTION
**Pull request by bkabrda**
### What does this PR do?

This PR implements signing individual DEB packages. It depends on https://github.com/DataDog/omnibus-ruby/pull/112, hence the temporary change to `release.json`.

In order to verify the package signature `debsig-verify`, users have to create two files on their filesystems (note that `4B4593018387EEAF` is the fingerprint of our signing key, which is a subkey of [1]):

* `/usr/share/debsig/keyrings/4B4593018387EEAF/debsig.gpg` - this is the *binary* (not ASCII) public datadog APT signing key [1] (in order to get the binary version, you have to first import the ASCII version and then export it without the `--armor` argument)
* `/etc/debsig/policies/4B4593018387EEAF/datadog.pol` - this is a policy file used by dpkg or debsig-verify to verify the signature. The file contents might be something like this:

```xml
<?xml version="1.0"?>
<!DOCTYPE Policy SYSTEM "https://www.debian.org/debsig/1.0/policy.dtd">
<Policy xmlns="https://www.debian.org/debsig/1.0/">
	<Origin Name="Datadog" id="4B4593018387EEAF" Description="Datadog, Inc" />

	<Selection>
		<Required Type="origin" File="debsig.gpg" id="4B4593018387EEAF" />
	</Selection>

	<Verification MinOptional="0">
		<Required Type="origin" File="debsig.gpg" id="4B4593018387EEAF" />
	</Verification>
</Policy>
```

To actually verify the package, one can:

* Call `debsig-verify <path-to-deb-file>` manually on a downloaded package file.
* Modify `/etc/dpkg/dpkg.cfg` and comment out the line containing `no-debsig`. The unfortunate side effect of this is that `dpkg` will now start verifying signatures of *all* downloaded packages from *all* sources (there is no possibility to select sources which would be verified and which not). Because Debian doesn't sign the packages in their official sources, this will effectively make it impossible to install any package from the official sources.

[1] https://keyserver.ubuntu.com/pks/lookup?op=get&search=0xa2923dff56eda6e76e55e492d3a80e30382e94de

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Describe your test plan

Write there any instructions and details that should be tested during the QA.

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -135,6 +135,11 @@ variables:
   BCC_VERSION: v0.12.0
   SYSTEM_PROBE_GO_VERSION: 1.14.12
   DATADOG_AGENT_EMBEDDED_PATH: /opt/datadog-agent/embedded
+  # the deb signing key is split in halves, as it exceeds the maximum size of 4096 chars for a single entry
+  DEB_GPG_KEY_NAME: "Datadog, Inc <package@datadoghq.com>"
+  DEB_GPG_KEY_SSM_NAME_PART1: ci.datadog-agent.deb_signing_private_key_8387EEAF_part1
+  DEB_GPG_KEY_SSM_NAME_PART2: ci.datadog-agent.deb_signing_private_key_8387EEAF_part2
```
Comment (albertvaka) [inline .gitlab-ci.yml:141]: I wonder why we need to have it in two parts here but not in other places we use the key 🤔

**Review by albertvaka — APPROVED** [review]

File: .gitlab-ci.yml (hunk 1 above)
Comment (KSerrania) [reply .gitlab-ci.yml:141]: It depends on whether standard-size secrets or advanced-size secrets are used (https://docs.aws.amazon.com/systems-manager/latest/userguide/parameter-store-advanced-parameters.html).

For instance, the key used to sign the staging repos is stored in standard-size secrets, and thus is also split in two parts: https://github.com/DataDog/datadog-agent/blob/6275ef2adaf6dcacff9a3b0630d2e8ecf7762629/.gitlab/deploy_7/nix.yml#L36-L39

**Comment by bkabrda** [comment]
The gitlab pipeline is failing now because I reverted the change to `release.json` and didn't merge the omnibus PR yet. I'll rerun it after the omnibus PR gets merged to ensure nothing got broken.

**Comment by bkabrda** [comment]
CI is green, I'm merging this PR.

**Pull request merged**